core/src/nucleus/actions/build_validation_package.rs

```diff
@@ -332,13 +355,82 @@ mod tests {
         ));
         assert!(maybe_validation_package.is_ok());
 
+        let headers = all_chain_headers_before_header(&context, &chain_header);
+
         let expected = ValidationPackage {
             chain_header: chain_header,
-            source_chain_entries: Some(all_public_chain_entries(&context)),
-            source_chain_headers: Some(all_chain_headers(&context)),
+            source_chain_entries: Some(public_chain_entries_from_headers(&context, &headers)),
+            source_chain_headers: Some(headers),
             custom: None,
         };
 
         assert_eq!(maybe_validation_package.unwrap(), expected);
     }
+
+    // test can make validation package with empty chain
+    #[test]
+    fn test_all_chain_headers_before_header_empty_chain() {
+        let (_instance, context) = instance(None);
+        let top_header = context
+            .state()
+            .unwrap()
+            .agent()
+            .top_chain_header()
+            .expect("There must be a top chain header");
+        let headers = all_chain_headers_before_header(&context, &top_header);
+        assert_eq!(headers.len(), 1) // includes the DNA entry only (no agent entry)
+    }
+
+    #[test]
+    fn test_all_chain_headers_before_header_entry_local_commit_validation() {
+        let (_instance, context) = instance(None);
+
+        let top_header = context
+            .state()
+            .unwrap()
+            .agent()
+            .top_chain_header()
+            .expect("There must be a top chain header");
+        // new entry header is created so it points to previous top header but not added to the local chain
+        let new_entry_header = ChainHeader::new(
+            &EntryType::from("test-new-entry"),
+            &Address::from("Qmtestaddress"),
+            &Vec::new(),
+            &Some(top_header.address()),
+            &None,
+            &None,
+            &Iso8601::new(0, 0),
+        );
+
+        let headers = all_chain_headers_before_header(&context, &new_entry_header);
+        // entry should not appear in the validating chain
+        assert_eq!(headers.contains(&new_entry_header), false);
+        assert_eq!(headers.len(), 2) // includes the DNA and agent entries
+    }
+
+    #[test]
+    fn test_all_chain_headers_before_header_entry_dht_validation() {
+        let (_instance, context) = instance(None);
+        // entry is added to the local chain
+        let chain_header = commit(test_entry_package_chain_full(), &context);
+        let headers = all_chain_headers_before_header(&context, &chain_header);
+        // entry should not appear in the validating chain
+        assert_eq!(headers.contains(&chain_header), false);
+        assert_eq!(headers.len(), 2) // includes the DNA and agent entries
```

Comment:
Do we actually include the DNA in validation packages? But we are at least filtering out the WASM, no? Valiation packages would be way to big otherwise...

Reply:
I think we do with the data redacted.